### PR TITLE
Cartesian boundary grid

### PR DESCRIPTION
**Pull request by oriolcg**
This PR enables to explicitly assume an affine map in the construction of a grid, a feature used in the construction of grids from `CartesianDiscreteModel`.

### CODE REVIEW

File: src/Geometry/UnstructuredGrids.jl [hunk 1]
```diff
@@ -32,9 +32,15 @@ struct UnstructuredGrid{Dc,Dp,Tp,O,Tn} <: Grid{Dc,Dp}
...
-
-    cell_map = _compute_cell_map(node_coordinates,cell_node_ids,reffes,cell_types)
+    facet_normal=nothing;
+    has_affine_map=nothing) where {Dc,Dp,Tp,Ti}
+
+    if has_affine_map === nothing
+      _has_affine_map = get_has_affine_map(reffes)
+      cell_map = _compute_cell_map(node_coordinates,cell_node_ids,reffes,cell_types,_has_affine_map)
+    else
+      cell_map = _compute_cell_map(node_coordinates,cell_node_ids,reffes,cell_types,has_affine_map)
```
Comment (fverdugo) [inline src/Geometry/UnstructuredGrids.jl:42]: else
  _has_affine_map = has_affine_map
end
cell_map = _compute_cell_map(...

Comment (fverdugo) [reply src/Geometry/UnstructuredGrids.jl:42]: I.e move the computation of cell_map outside the if

File: test/tmp.jl [hunk 2]
```diff
@@ -0,0 +1,62 @@
+using Pkg
+Pkg.activate(".")
```
[the remark this answers is not in the document]
Comment (oriolcg) [reply test/tmp.jl]: done in d0e67e616209fa5e8f2ddd652442e2081deb3ff5

File: src/Geometry/UnstructuredGrids.jl (hunk 1 above)
Comment (oriolcg) [reply src/Geometry/UnstructuredGrids.jl]: done in d0e67e616209fa5e8f2ddd652442e2081deb3ff5

**Review by fverdugo — APPROVED** [review]

**Pull request merged**